src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = Figures.values().length;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE_COUNT);
+        switch (index) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
+                        getRandomSide());
```

Comment:
lets add private methods to get each figure - to make your code easier to read
```suggestion
                return getRandomSquare();
```